doc/README.md

```diff
@@ -0,0 +1,83 @@
+# Network Distributed Architecture
+
+## Websockets - Default Response for New Connection
+
+```mermaid
+sequenceDiagram
+    autonumber
+    box Process Id p1
+    participant runtime@p1   
+    end
+
+    box Process Id p2
+    participant webgui@p2
+    participant runtime@p2
+    end
+
+    Note right of runtime@p1: Client runtime@p1 opens a <br/>websocket to remote webgui
+
+    runtime@p1->>webgui@p2: Connect<br/>ws://localhost:8888/api/messages?user=root&pwd=pwd&session_id=2309adf3dlkdk&id=p1
+
+    Note left of webgui@p2: Remote webgui@p2 and runtime@p2 attempt to subscribe <br/> to the describe method of the runtime@p1
+    webgui@p2->>runtime@p1: addListener describe
+    Note left of webgui@p2: runtime@p2 sends a describe request to runtime@p1
+    webgui@p2->>runtime@p1: describe 
+    Note right of runtime@p1: runtime@p1 responds with a describe response
+    runtime@p1->>webgui@p2: onDescribe 
+    Note left of webgui@p2: Based on the results of the describe,<br/> more querying and subscriptions can be processed
+
+
+    %% opt Server Add Listener, Describe and Reserve
+    %% end
+
+
+```
+### 1 Connection
+A connection with a websocket starts with an HTTP GET 
+```
+ws://localhost:8888/api/messages?user=root&pwd=pwd&session_id=2309adf3dlkdk&id=p1
+```
+### 2 Subscribe to Runtime Describe
+When a connection is established between two different myrobotlab instances p1 and p2,
+the following messages are sent.  The first message is adding a subscription to the runtime of the
+other process.  The subscription is for the function "describe".  The second message will be a describe request.
+```json
+{
+  "msgId": 1690145377501,
+  "name": "runtime",
+  "sender": "webgui@p2",
+  "sendingMethod": "",
+  "historyList": [],
+  "properties": null,
+  "status": null,
+  "encoding": "json",
+  "method": "addListener",
+  "data": [
+    "{\"topicMethod\":\"describe\",\"callbackName\":\"runtime@caring-hector\",\"callbackMethod\":\"onDescribe\",\"class\":\"org.myrobotlab.framework.MRLListener\"}"
+  ],
+  "class": "org.myrobotlab.framework.Message"
+}
+```
+### 3 Send Describe
```

Comment:
Should explain what the describe message is actually for. Notably, what is "fill-uuid" and why is it important? Should specify the stage at which a Runtime instance is `register()`ed, which I believe is in the receiving end of this message

Reply:
updated, fill-uuid should be refactored out.  There should be more simple set of criteria and responses, starting wtih just a list of local services and their names & "types".. although "Type" information is of limited use.  To be polyglot, it really should be the ability to get itnterface information which parameters in the form of openapi schema